### PR TITLE
feat(pie-docs): DSW-1324 add column component

### PR DESCRIPTION
**Pull request by FayeCarter**
## Author Checklist (complete before requesting a review)
- [ ] I have performed a self-review of my code
- [ ] If it is a core feature, I have added thorough tests
- [ ] If it is a `PIE Docs` change, I have reviewed the Docs site preview
- [ ] If it is a component change, I have reviewed the Storybook preview
- [ ] If there are visual test updates, I have reviewed them properly before approving

### Reviewer 1
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### Reviewer 2
- [ ] If it is a `PIE Docs` change, I have reviewed the PR preview
- [ ] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: apps/pie-docs/src/components/template/template.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
   "title": "Template",
   "navKey": "Template",
-  "description": "[Component description]",
-  "draft": true
```
Comment (FayeCarter) [inline apps/pie-docs/src/components/template/template.json:5]: this has just been done for a design review, I'll add it back in

File: apps/pie-docs/src/components/template/code.md [hunk 2]
```diff
@@ -6,3 +6,74 @@ eleventyNavigation:
 ---
 
 ## Code
+
+{% contentLayout %}
```
Comment (FayeCarter) [inline apps/pie-docs/src/components/template/code.md:10]: This is just for design review, I'll add it back in

File: apps/pie-docs/src/_11ty/shortcodes/contentLayout.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+const markdownFilter = require('../filters/markdown');
+const { deindentHTML } = require('./shortcode-utilities');
+const headingAnchor = require('../filters/headingAnchor');
+
+/**
+ * Creates an container for a selection of elements that dhould be displayed in one column.
```
Comment (raoufswe) [inline apps/pie-docs/src/_11ty/shortcodes/contentLayout.js:6]:
```suggestion
 * Creates a container for a selection of elements that should be displayed in one column.
```

File: apps/pie-docs/src/_11ty/shortcodes/contentLayout.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * @param {object} config.content - content to be diplayed in columns
```
Comment (raoufswe) [inline apps/pie-docs/src/_11ty/shortcodes/contentLayout.js:7]:
```suggestion
 * @param {object} content - content to be displayed in columns
```

File: apps/pie-docs/src/_11ty/shortcodes/contentLayout.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+ */
+const contentItem = (content) => `<div class="c-columns-item">
+        ${markdownFilter(deindentHTML(content))}
+    </div>`;
+
+/**
+ * Creates an container that displays contents in two columns.
```
Comment (raoufswe) [inline apps/pie-docs/src/_11ty/shortcodes/contentLayout.js:14]:
```suggestion
 * Creates a container that displays contents in two columns.
```

File: apps/pie-docs/src/assets/styles/components/_contentImage.scss [hunk 6]
```diff
@@ -1,18 +1,47 @@
 @use '../fozzie-config' as f;
 @use '../variables' as v;
 
-$contentImage-bg-minHeight:         280px;
+$contentImage-height:                     200px;
+$contentImage-height--mobile:             160px;
+$contentImage-height-m:                   400px;
+$contentImage-height-m--mobile:           240px;
+$contentImage-height-l:                   600px;
+$contentImage-height-l--mobile:           400px;
```
Comment (raoufswe) [inline apps/pie-docs/src/assets/styles/components/_contentImage.scss:9]: where are these values coming from?

File: apps/pie-docs/src/_11ty/shortcodes/contentLayout.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+ * @param {object} config.content - content to be diplayed in columns
```
Comment (xander-marjoram) [inline apps/pie-docs/src/_11ty/shortcodes/contentLayout.js:15]:
```suggestion
 * @param {object} content - content to be displayed in columns
```

File: apps/pie-docs/src/_11ty/shortcodes/contentPageImage.js [hunk 8]
```diff
@@ -15,6 +15,7 @@ const createCaption = (config) => (config.caption
  * If a width is not provided, image will default to fill up the entire container.
  * @param {object} config - image configuration
  * @param {string} config.width - the image width in px (will go to full width on smaller screens).
+ * @param {string} config.height - (s, m, l), the image height in px will default to small.
```
Comment (xander-marjoram) [inline apps/pie-docs/src/_11ty/shortcodes/contentPageImage.js:18]: `in px` seems a little misleading if we're asking for `s`/`m`/`l`

File: apps/pie-docs/src/assets/styles/base/_layout.scss [hunk 9]
```diff
@@ -90,6 +90,14 @@
         padding-block-start: f.spacing(f);
         padding-block-end: f.spacing(j);
 
+        @include f.media('>=wide') {
+            padding-top: f.spacing(h);
+            padding-bottom: calc(f.spacing(j) + f.spacing(d));
```
[the remark this answers is not in the document]
Comment (xander-marjoram) [reply apps/pie-docs/src/assets/styles/base/_layout.scss:95]: I agree it's worth a try but I think we need a wider ticket for updating these across the docs site

File: apps/pie-docs/src/assets/styles/components/_contentImage.scss (hunk 6 above)
Comment (FayeCarter) [reply apps/pie-docs/src/assets/styles/components/_contentImage.scss:9]: Naz has requested them

**Review by raoufswe — APPROVED** [review]

**Review by xander-marjoram — APPROVED** [review]

**Pull request merged**